### ISSUE TITLE
Running editor crashes after asset processor finishes.

### ISSUE THREAD

**Issue by michleza** [o3de/o3de#9418]
**Describe the bug**
If you attempt to run latest dev there are numerous breaks during asset processing then crashes after the asset processor has finished

**Steps to reproduce**
1. pull latest development
2. configure and generate in cmake
3. build in visual studio
4. run editor as your startup project
5. wait for asset processor to finish. click continue through any break mode stops you encounter

**Expected behavior**
running editor will trigger asset processor then launch editor.

**Actual behavior**
Exceptions thrown in cpp file: ShaderResourceGroupLayout.cpp

**Screenshots/Video**
![image](https://user-images.githubusercontent.com/98843754/167042585-3ffe4341-873c-46c1-88a6-3f7af5086532.png)

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
 - Device: win 10
 - CPU Ryzen 5950x, ]
 - GPUNVidia RTX 3070]
 - Memory [ 32GB]
 - 
[editorCallStack.txt](https://github.com/o3de/o3de/files/8636611/editorCallStack.txt)

**Comment by michleza** [comment]
This blocks usage of the editor.

**Comment by AMZN-stankowi** [comment]
I don't think this is totally related, but I'm looking at https://github.com/o3de/o3de/issues/9387 let me know if it looks like 9418 is caused by an asset hang similar to 9387

**Comment by michleza** [comment]
> I don't think this is totally related, but I'm looking at #9387 let me know if it looks like 9418 is caused by an asset hang similar to 9387

It may be the case. If I run the executable instead of running from VS then Asset processor hangs for a long time. Afterwards everything besides project manager closes without any warnings or anything

**Comment by santorac** [comment]
I am not able to repro this issue. Is it still occurring?

Is it possible that when this was happening, you didn't keep the Editor.exe as the focused process, but had focus on some other window? That was known to cause some asset loading edge cases. It's just a hunch, but maybe it was fixed by https://github.com/o3de/o3de/commit/43b35f6a3103072d9b7a88871a25c4ab2c6dbbe8

If it is still occurring, can you work around the issue by waiting for the Asset Processor to finish before starting the Editor?

Also, it would be helpful to see the log. In the first callstack the failure looks like RPISystemInterface::GetViewSrgLayout is returning null. From what I see in the code, there should be some error messages leading up to this failure, which would be helpful to see. 

I'm not sure how GetViewSrgLayout would return null, since RPISystem::InitializeSystemAssets() calls LoadCriticalAsset which should block until the asset is available and ready. Also CoreLightsSystemComponent and BootstrapSystemComponent both list "RPISystem" as a required service, so the RPI should be initialized before BootstrapSystemComponent attempts to call RPISystem::InitializeSystemAssets().

Are there any AP errors related to SceneAndViewSrgs.shader?

(I noticed there are two callstacks linked above, and they are pretty different from each other. I'm not sure they represent the same issue).

**Comment by santorac** [comment]
I think I just hit this issue with the latest development branch. I can't be sure because the OP didn't include a callstack, but the repro steps are the same and both are a null pointer crash in ShaderResourceGroupLayout. I had just synced, ran the Editor, the AP had a bunch of processing to do, then the Editor crashed with a null pointer access.

Callstack:
```
>	Atom_RPI.Editor.dll!AZ::RPI::ShaderOptionGroupLayout::FindShaderOptionIndex(const AZ::Name & optionName) Line 558	C++
 	Atom_RPI.Editor.dll!AZ::RPI::LuaMaterialFunctorRuntimeContext::SetShaderOptionValueHelper(const char * name, AZStd::function<bool __cdecl(AZ::RPI::ShaderOptionGroup *,AZ::RHI::Handle<unsigned int,AZ::RPI::ShaderOptionIndexNamespace>)> setValueCommand) Line 391	C++
 	Atom_RPI.Editor.dll!AZ::RPI::LuaMaterialFunctorRuntimeContext::SetShaderOptionValue<bool>(const char * name, bool value) Line 427	C++
 	Atom_RPI.Editor.dll!AZ::Internal::CallFunction<bool,char const *,bool>::Member<0,1,AZ::RPI::LuaMaterialFunctorRuntimeContext *,bool (__cdecl AZ::RPI::LuaMaterialFunctorRuntimeContext::*)(char const *,bool)>(bool(AZ::RPI::LuaMaterialFunctorRuntimeContext::*)(const char *, bool) functionPtr, AZ::RPI::LuaMaterialFunctorRuntimeContext * thisPtr, AZ::BehaviorArgument * arguments, AZ::BehaviorArgument * result, std::integer_sequence<unsigned __int64,0,1> __formal) Line 4045	C++
 	Atom_RPI.Editor.dll!AZ::Internal::BehaviorMethodImpl<bool (__cdecl AZ::RPI::LuaMaterialFunctorRuntimeContext::*)(char const *,bool)>::Call(AZ::BehaviorArgument * arguments, unsigned int numArguments, AZ::BehaviorArgument * result) Line 4318	C++
 	Atom_RPI.Editor.dll!AZ::LuaScriptCaller::Call(lua_State * lua) Line 3559	C++
 	[External Code]	
 	Atom_RPI.Editor.dll!AZ::Internal::LuaSafeCall(lua_State * lua, int numParams, int numResults) Line 616	C++
 	Atom_RPI.Editor.dll!AZ::ScriptDataContext::CallExecute() Line 2185	C++
 	Atom_RPI.Editor.dll!AZ::RPI::LuaMaterialFunctor::Process(AZ::RPI::MaterialFunctor::RuntimeContext & context) Line 135	C++
 	Atom_RPI.Editor.dll!AZ::RPI::Material::Compile() Line 375	C++
 	Atom_RPI.Editor.dll!AZ::RPI::Material::Init(AZ::RPI::MaterialAsset & materialAsset) Line 158	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZ::RPI::Material::CreateInternal(AZ::RPI::MaterialAsset & materialAsset) Line 47	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZ::RPI::MaterialSystem::Init::__l2::<lambda_1>::operator()(AZ::Data::AssetData *) Line 46	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::Internal::INVOKE(AZ::RPI::MaterialSystem::Init::__l2::<lambda_1> &) Line 208	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::invoke(AZ::RPI::MaterialSystem::Init::__l2::<lambda_1> &) Line 54	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::Internal::function_util::invoke_void_return_wrapper<AZStd::intrusive_ptr<AZ::RPI::Material>>::call(AZ::RPI::MaterialSystem::Init::__l2::<lambda_1> &) Line 32	C++
 	Atom_RPI.Editor.dll!AZStd::Internal::function_util::get_invoker<AZStd::intrusive_ptr<AZ::RPI::Material> __cdecl(AZ::Data::AssetData *),AZStd::Internal::function_util::function_obj_tag,AZStd::allocator>::call<`AZ::RPI::MaterialSystem::Init'::`2'::<lambda_1>>(AZStd::Internal::function_util::function_buffer & function_obj_ptr, AZ::Data::AssetData * && <args_0>) Line 172	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::function_intermediate<AZStd::intrusive_ptr<AZ::RPI::Material>,AZ::Data::AssetData *>::operator()(AZ::Data::AssetData * &&) Line 604	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::function<AZStd::intrusive_ptr<AZ::RPI::Material> __cdecl(AZ::Data::AssetData *)>::operator()(AZ::Data::AssetData * <args_0>) Line 684	C++
 	Atom_RPI.Editor.dll!AZ::Data::InstanceDatabase<AZ::RPI::Material>::FindOrCreate(const AZ::Data::InstanceId & id, const AZ::Data::Asset<AZ::Data::AssetData> & asset, const AZStd::any * param) Line 340	C++
 	Atom_RPI.Editor.dll!AZ::RPI::Material::FindOrCreate(const AZ::Data::Asset<AZ::RPI::MaterialAsset> & materialAsset) Line 35	C++
 	Atom_RPI.Editor.dll!AZ::RPI::ModelLod::Init(const AZ::Data::Asset<AZ::RPI::ModelLodAsset> & lodAsset, const AZ::Data::Asset<AZ::RPI::ModelAsset> & modelAsset) Line 115	C++
 	Atom_RPI.Editor.dll!AZ::RPI::ModelLod::CreateInternal(const AZ::Data::Asset<AZ::RPI::ModelLodAsset> & lodAsset, const AZStd::any * modelAssetAny) Line 44	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZ::RPI::ModelSystem::Init::__l2::<lambda_1>::operator()(AZ::Data::AssetData *) Line 48	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::Internal::INVOKE(AZ::RPI::ModelSystem::Init::__l2::<lambda_1> &) Line 208	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::invoke(AZ::RPI::ModelSystem::Init::__l2::<lambda_1> &) Line 54	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::Internal::function_util::invoke_void_return_wrapper<AZStd::intrusive_ptr<AZ::RPI::ModelLod>>::call(AZ::RPI::ModelSystem::Init::__l2::<lambda_1> &) Line 32	C++
 	Atom_RPI.Editor.dll!AZStd::Internal::function_util::get_invoker<AZStd::intrusive_ptr<AZ::RPI::ModelLod> __cdecl(AZ::Data::AssetData *,AZStd::any const *),AZStd::Internal::function_util::function_obj_tag,AZStd::allocator>::call<`AZ::RPI::ModelSystem::Init'::`2'::<lambda_1>>(AZStd::Internal::function_util::function_buffer & function_obj_ptr, AZ::Data::AssetData * && <args_0>, const AZStd::any * && <args_1>) Line 172	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::function_intermediate<AZStd::intrusive_ptr<AZ::RPI::ModelLod>,AZ::Data::AssetData *,AZStd::any const *>::operator()(AZ::Data::AssetData * &&) Line 604	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::function<AZStd::intrusive_ptr<AZ::RPI::ModelLod> __cdecl(AZ::Data::AssetData *,AZStd::any const *)>::operator()(AZ::Data::AssetData * <args_0>, const AZStd::any * <args_1>) Line 684	C++
 	Atom_RPI.Editor.dll!AZ::Data::InstanceDatabase<AZ::RPI::ModelLod>::FindOrCreate(const AZ::Data::InstanceId & id, const AZ::Data::Asset<AZ::Data::AssetData> & asset, const AZStd::any * param) Line 344	C++
 	Atom_RPI.Editor.dll!AZ::RPI::ModelLod::FindOrCreate(const AZ::Data::Asset<AZ::RPI::ModelLodAsset> & lodAsset, const AZ::Data::Asset<AZ::RPI::ModelAsset> & modelAsset) Line 25	C++
 	Atom_RPI.Editor.dll!AZ::RPI::Model::Init(const AZ::Data::Asset<AZ::RPI::ModelAsset> & modelAsset) Line 96	C++
 	Atom_RPI.Editor.dll!AZ::RPI::Model::CreateInternal(const AZ::Data::Asset<AZ::RPI::ModelAsset> & modelAsset) Line 71	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZ::RPI::ModelSystem::Init::__l2::<lambda_2>::operator()(AZ::Data::AssetData *) Line 56	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::Internal::INVOKE(AZ::RPI::ModelSystem::Init::__l2::<lambda_2> &) Line 208	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::invoke(AZ::RPI::ModelSystem::Init::__l2::<lambda_2> &) Line 54	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::Internal::function_util::invoke_void_return_wrapper<AZStd::intrusive_ptr<AZ::RPI::Model>>::call(AZ::RPI::ModelSystem::Init::__l2::<lambda_2> &) Line 32	C++
 	Atom_RPI.Editor.dll!AZStd::Internal::function_util::get_invoker<AZStd::intrusive_ptr<AZ::RPI::Model> __cdecl(AZ::Data::AssetData *),AZStd::Internal::function_util::function_obj_tag,AZStd::allocator>::call<`AZ::RPI::ModelSystem::Init'::`2'::<lambda_2>>(AZStd::Internal::function_util::function_buffer & function_obj_ptr, AZ::Data::AssetData * && <args_0>) Line 172	C++
 	[Inline Frame] DiffuseProbeGrid.Editor.dll!AZStd::function_intermediate<AZStd::intrusive_ptr<AZ::RPI::Model>,AZ::Data::AssetData *>::operator()(AZ::Data::AssetData * &&) Line 604	C++
 	[Inline Frame] DiffuseProbeGrid.Editor.dll!AZStd::function<AZStd::intrusive_ptr<AZ::RPI::Model> __cdecl(AZ::Data::AssetData *)>::operator()(AZ::Data::AssetData * <args_0>) Line 684	C++
 	DiffuseProbeGrid.Editor.dll!AZ::Data::InstanceDatabase<AZ::RPI::Model>::FindOrCreate(const AZ::Data::InstanceId & id, const AZ::Data::Asset<AZ::Data::AssetData> & asset, const AZStd::any * param) Line 340	C++
 	DiffuseProbeGrid.Editor.dll!AZ::RPI::Model::FindOrCreate(const AZ::Data::Asset<AZ::RPI::ModelAsset> & modelAsset) Line 28	C++
 	DiffuseProbeGrid.Editor.dll!AZ::Render::DiffuseProbeGridFeatureProcessor::OnVisualizationModelAssetReady(AZ::Data::Asset<AZ::Data::AssetData> asset) Line 837	C++
 	DiffuseProbeGrid.Editor.dll!AZ::Render::DiffuseProbeGridFeatureProcessor::OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData> asset) Line 924	C++
 	[Inline Frame] EditorLib.dll!AZStd::Internal::INVOKE(void(AZ::Data::AssetEvents::*)(AZ::Data::Asset<AZ::Data::AssetData>) &) Line 177	C++
 	[Inline Frame] EditorLib.dll!AZStd::invoke(void(AZ::Data::AssetEvents::*)(AZ::Data::Asset<AZ::Data::AssetData>) &) Line 54	C++
 	[Inline Frame] EditorLib.dll!AZ::EBusEventProcessingPolicy::Call(void(AZ::Data::AssetEvents::*)(AZ::Data::Asset<AZ::Data::AssetData>) &) Line 437	C++
 	EditorLib.dll!AZ::Internal::EBusContainer<AZ::Data::AssetEvents,AZ::Data::AssetEvents,1,1>::Dispatcher<AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents>>::Event<void (__cdecl AZ::Data::AssetEvents::*)(AZ::Data::Asset<AZ::Data::AssetData>),AZ::Data::Asset<AZ::Data::AssetData> &>(const AZ::Data::AssetId & id, void(AZ::Data::AssetEvents::*)(AZ::Data::Asset<AZ::Data::AssetData>) && func, AZ::Data::Asset<AZ::Data::AssetData> & <args_0>) Line 146	C++
 	EditorLib.dll!AZ::Data::AssetManager::NotifyAssetReady(AZ::Data::Asset<AZ::Data::AssetData> asset) Line 1836	C++
 	[Inline Frame] EditorLib.dll!AZStd::Internal::INVOKE(void(AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>) &&) Line 177	C++
 	[Inline Frame] EditorLib.dll!AZStd::invoke(void(AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>) &&) Line 54	C++
 	[Inline Frame] EditorLib.dll!AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents>,AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents,AZ::Data::AssetEvents>>::QueueFunction::__l5::<lambda_1>::operator()() Line 860	C++
 	[Inline Frame] EditorLib.dll!AZStd::Internal::INVOKE(AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents>,AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents,AZ::Data::AssetEvents>>::QueueFunction::__l5::<lambda_1> &) Line 208	C++
 	[Inline Frame] EditorLib.dll!AZStd::invoke(AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents>,AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents,AZ::Data::AssetEvents>>::QueueFunction::__l5::<lambda_1> &) Line 54	C++
 	[Inline Frame] EditorLib.dll!AZStd::Internal::function_util::invoke_void_return_wrapper<void>::call(AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents>,AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents,AZ::Data::AssetEvents>>::QueueFunction::__l5::<lambda_1> &) Line 41	C++
 	EditorLib.dll!AZStd::Internal::function_util::get_invoker<void __cdecl(void),AZStd::Internal::function_util::function_obj_tag,AZ::Internal::EBusEnvironmentAllocator>::call<`AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents>,AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents,AZ::Data::AssetEvents>>::QueueFunction<void (__cdecl AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>),AZ::Data::AssetManager *,AZ::Data::Asset<AZ::Data::AssetData>>'::`5'::<lambda_1>>(AZStd::Internal::function_util::function_buffer & function_obj_ptr) Line 148	C++
 	[Inline Frame] Atom_Feature_Common.Editor.dll!AZStd::function_intermediate<void>::operator()() Line 604	C++
 	[Inline Frame] Atom_Feature_Common.Editor.dll!AZStd::function<void __cdecl(void)>::operator()() Line 684	C++
 	Atom_Feature_Common.Editor.dll!AZ::EBusQueuePolicy<1,AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents>,AZStd::recursive_mutex>::Execute() Line 267	C++
 	[Inline Frame] Atom_Feature_Common.Editor.dll!AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents>,AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents,AZ::Data::AssetEvents>>::ExecuteQueuedEvents() Line 366	C++
 	Atom_Feature_Common.Editor.dll!AZ::Data::AssetManager::DispatchEvents() Line 591	C++
 	Atom_Feature_Common.Editor.dll!AZ::Data::WaitForAsset::Wait() Line 339	C++
 	Atom_Feature_Common.Editor.dll!AZ::Data::WaitForAsset::WaitUntilReady() Line 323	C++
 	Atom_Feature_Common.Editor.dll!AZ::Data::AssetManager::BlockUntilLoadComplete(const AZ::Data::Asset<AZ::Data::AssetData> & asset) Line 849	C++
 	Atom_Feature_Common.Editor.dll!AZ::Data::Asset<AZ::RPI::ShaderAsset>::BlockUntilLoadComplete() Line 1153	C++
 	Atom_Feature_Common.Editor.dll!AZ::RPI::FindShaderAsset(AZ::Data::AssetId shaderAssetId, const AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> & shaderFilePath) Line 362	C++
 	Atom_Feature_Common.Editor.dll!AZ::RPI::LoadShader(AZ::Data::AssetId shaderAssetId, const AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> & shaderFilePath, const AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> & supervariantName) Line 374	C++
 	Atom_Feature_Common.Editor.dll!AZ::RPI::LoadCriticalShader(const AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> & shaderFilePath, const AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> & supervariantName) Line 409	C++
 	Atom_Feature_Common.Editor.dll!AZ::Render::DynamicPrimitiveProcessor::InitShader() Line 323	C++
 	Atom_Feature_Common.Editor.dll!AZ::Render::DynamicPrimitiveProcessor::Initialize(const AZ::RPI::Scene * scene) Line 54	C++
 	Atom_Feature_Common.Editor.dll!AZ::Render::AuxGeomFeatureProcessor::Activate() Line 45	C++
 	Atom_Bootstrap.dll!AZ::RPI::Scene::Activate() Line 151	C++
 	Atom_Bootstrap.dll!AZ::Render::Bootstrap::BootstrapSystemComponent::GetOrCreateAtomSceneFromAzScene(AzFramework::Scene * scene) Line 281	C++
 	EditorLib.dll!`AtomToolsFramework::RenderViewportWidget::SetScene'::`2'::<lambda_1>::operator()(AZ::Render::Bootstrap::Request * bootstrapRequests) Line 166	C++
 	[Inline Frame] EditorLib.dll!AZStd::Internal::INVOKE(AtomToolsFramework::RenderViewportWidget::SetScene::__l2::<lambda_1> &) Line 208	C++
 	[Inline Frame] EditorLib.dll!AZStd::invoke(AtomToolsFramework::RenderViewportWidget::SetScene::__l2::<lambda_1> &) Line 54	C++
 	[Inline Frame] EditorLib.dll!AZ::EBusEventProcessingPolicy::Call(AtomToolsFramework::RenderViewportWidget::SetScene::__l2::<lambda_1> &) Line 437	C++
 	EditorLib.dll!AZ::Internal::EBusContainer<AZ::Render::Bootstrap::Request,AZ::Render::Bootstrap::Request,0,0>::Dispatcher<AZ::EBus<AZ::Render::Bootstrap::Request,AZ::Render::Bootstrap::Request>>::Broadcast<`AtomToolsFramework::RenderViewportWidget::SetScene'::`2'::<lambda_1> &>(AtomToolsFramework::RenderViewportWidget::SetScene::__l2::<lambda_1> & func) Line 1541	C++
 	EditorLib.dll!AtomToolsFramework::RenderViewportWidget::SetScene(const AZStd::shared_ptr<AzFramework::Scene> & scene, bool useDefaultRenderPipeline) Line 175	C++
 	EditorLib.dll!EditorViewportWidget::UpdateScene() Line 2432	C++
 	EditorLib.dll!EditorViewportWidget::SetViewportId(int id) Line 925	C++
 	EditorLib.dll!CLayoutViewPane::AttachViewport(QWidget * pViewport) Line 304	C++
 	EditorLib.dll!CLayoutViewPane::SetViewClass(const QString & sClass) Line 220	C++
 	[Inline Frame] EditorLib.dll!CLayoutWnd::BindViewport(CLayoutViewPane * vp, const QString &) Line 165	C++
 	EditorLib.dll!CLayoutWnd::MaximizeViewport(int paneId) Line 205	C++
 	EditorLib.dll!CLayoutWnd::LoadConfig() Line 515	C++
 	EditorLib.dll!MainWindow::InitCentralWidget() Line 444	C++
 	EditorLib.dll!MainWindow::Initialize() Line 469	C++
 	EditorLib.dll!CCryEditApp::InitInstance() Line 1745	C++
 	EditorLib.dll!CryEditMain(int argc, char * * argv) Line 4011	C++
 	[Inline Frame] Editor.exe!AZStd::Internal::INVOKE(int(*)(int, char * *) &) Line 208	C++
 	[Inline Frame] Editor.exe!AZStd::invoke(int(*)(int, char * *) &) Line 54	C++
 	Editor.exe!main(int argc, char * * argv) Line 32	C++
 	Editor.exe!WinMain(HINSTANCE__ * __formal, HINSTANCE__ * __formal, char * __formal, int __formal) Line 97	C++
 	[External Code]	
```

Analysis:
- ShaderOptionGroupLayout::FindShaderOptionIndex - _this_ is null
- LuaMaterialFunctorRuntimeContext::SetShaderOptionValueHelper 
    - The ShaderOptionGroup appears to be valid memory, but is maybe default initialized. It's m_layout is null and m_id is default, all 0's. 
    - The ShaderCollection::Item is valid: "DepthPass" shader asset "shaders/depth/depthpass.azshader"
    - The ShaderCollection is obviously from StandardPBR, based on the shaders listed.
    - Some shaders such as the foward pass shaders have a valid ShaderOptionGroupLayout
    - Some of the simple utility shaders like the depth and shadow pass shaders have a null ShaderOptionGroupLayout. This makes me wonder it's valid to have no ShaderOptionGroupLayout if there are no shader options.
-

**Comment by santorac** [comment]
Note that both callstacks https://github.com/o3de/o3de/issues/9418#issuecomment-1143857381 and https://github.com/o3de/o3de/issues/9418#issuecomment-1244847072 are under DiffuseProbeGridFeatureProcessor::OnVisualizationModelAssetReady where it tries to create the probe visualization model. It appears to crash when trying to load the model's built-in material. I'm pretty sure the diffuse probe system doesn't actually use this material for rendering, it replaces it with a different one, so one possible workaround for this crash might be to just ensure that model does not have any built-in material.

**Comment by santorac** [comment]
I think we need to add more logging to a few places to get more detail about why asset loading failed, but I'm not sure exactly where. Maybe someone from @o3de/sig-core can help?

**Comment by santorac** [comment]
Running the Editor again after the AP has finished processing worked fine. @galibzon I recommend we reduce the priority of this bug, I don't think it's a blocker since it happens rarely, and when it does you just need to restart the Editor.

**Comment by lawsonamzn** [comment]
on step 5 of repro
wait for asset processor to finish. click continue through any break mode stops you encounter

There aren't supposed to be any break mode stops in AP or other crash issues during processing, so this might also be a hint.

**Comment by santorac** [comment]
>There aren't supposed to be any break mode stops in AP or other crash issues during processing, so this might also be a hint.

There might be some deviation from the original post, maybe even a completely separate issue. The crash that everyone is seeing now is immediate (once processing is done), there are no break mode stops to skip through.

**Comment by santorac** [comment]
I just hit a null ShaderResourceGroup crash again. This time it happened when the Editor booted while assets were still being processed, rather than happening when asset processing finished. I wonder if we can just mark the products of DiffuseProbeSphere.fbx as critical assets.

**Comment by galibzon** [comment]
I tried to reproduce this bug on both Windows and Linux using O3DE Version
```
commit 241717596c9db81c529a8581c98e856f036ee75a (HEAD -> development, upstream/development)
Author: michabr <82236305+michabr@users.noreply.github.com>
Date:   Thu Feb 2 10:46:53 2023 -0800

    Move Linux workaround for UI Editor (#14330)
    
    Signed-off-by: michabr <82236305+michabr@users.noreply.github.com>
```

Never got the crash. I tried two common scenarios:

1. Clean cache and start the Editor, which automatically starts the AP. the Editor waited until the AP completed and when finished the Editor opened and I was able to load a few levels.
2. Same as before by I killed the Editor 3 times, at different stages of Asset Processing. When 1000 assets were pending completion I left the Editor running and it allowed me to open Levels as soon as the AP was done.

Closing as can not reproduce.
